src/reference/asciidoc/amqp.adoc

```diff
@@ -2863,6 +2870,61 @@ This header can be obtained by adding the `@Header(AmqpHeaders.LAST_IN_BATCH)` b
 The header is mapped from `MessageProperties.isLastInBatch()`.
 In addition, `AmqpHeaders.BATCH_SIZE` is populated with the size of the batch in every message fragment.
 
+In addition, a new property `consumerBatchEnabled` has been added to the `SimpleMessageListenerContainer`.
+When this is true, the container will create a batch of messages, up to `batchSize`; a partial batch is delivered if `receiveTimeout` elapses with no new messages arriving.
+If a producer-created batch is received, it is debatched and added to the consumer-side batch; therefore the actual number of messages delivered may exceed `batchSize`, which represents the number of messages received from the broker.
+`deBatchingEnabled` must be true when `consumerBatchEnabled` is true.
```

Comment:
Why can't this happen automaticaly?

Reply:
Because `batchListener` on the factory serves 2 purposes.

1. configures the `BatchMessagingMessageListenerAdapter`
2. sets deBatchingEnabled to false (for producer batches).

I was not comfortable with this either; I think I need to add `consumerBatchEnabled` as a first class property on the factory.

Will push another commit.